### PR TITLE
[couch-to-sql][Repeaters][Phase-2] Move reads and writes to SQL

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Product Description

This PR combines four PRs
PR 1 https://github.com/dimagi/commcare-hq/pull/32021
PR 2 https://github.com/dimagi/commcare-hq/pull/32042
PR 3 https://github.com/dimagi/commcare-hq/pull/32102
PR 4 https://github.com/dimagi/commcare-hq/pull/32131

These PRs have been individually reviewed. Now opening up this final PR for testing and QA
### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [ ] This PR can be reverted after deploy with no further considerations

The PR contains migrations that should be rolled back before rolling back the deploy. The migrations at this point do not affect the code in any way.

The revert process should be - 
- Roll back the migrations.

```
manage.py migrate repeaters 0007

```
- Merge the Revert the PR https://github.com/dimagi/commcare-hq/pull/32377
- Deploy

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by millerdev — APPROVED** [review]
Did not re-review, but approving so it is easy to see if any new commits are added.

**Review by millerdev — APPROVED** [review]

**Comment by millerdev** [comment]
I think https://github.com/dimagi/commcare-hq/pull/32160/commits/f854403d2809362557c1646856fa157aca3c5a41 would not be necessary (because of https://github.com/dimagi/commcare-hq/pull/32184) if master was merged into this branch.

**Comment by kaapstorm** [comment]
Thanks @millerdev that explains it!

Before we merge master (which would also have made that other cherrypick unnecessary :upside_down_face: ), let me revert that commit: https://github.com/dimagi/commcare-hq/pull/32200

**Review by millerdev — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Comment by AmitPhulera** [comment]
Stickler is failing on the PR with `Timed out cloning repository` which I am ignoring for now, as it is a known [issue](https://dimagi.slack.com/archives/C02QG63MU/p1669704780509899)

File: corehq/motech/dhis2/entities_helpers.py
Comment (esoergel) [inline corehq/motech/dhis2/entities_helpers.py:201]: no idea whether this PR was the cause, but I just saw an alert for this bug:
https://sentry.io/organizations/dimagi/issues/3825537564/?project=136860&referrer=slack
And then saw that this PR recently changed the surrounding code, so I figured I'd flag in case that warrants investigation.

Comment (AmitPhulera) [reply corehq/motech/dhis2/entities_helpers.py:201]: Thanks for flagging it @esoergel. I will look into it.

**Pull request merged**
